vm/systemSmartContracts/governance.go

```diff
@@ -1005,13 +1038,40 @@ func (g *governanceContract) proposalContainsVoter(proposal *GeneralProposal, vo
 // computeVotingPower returns the voting power for a value. The value can be either a balance or
 //  the staked value for a validator
 func (g *governanceContract) computeVotingPower(value *big.Int) (*big.Int, error) {
-	if value.Cmp(big.NewInt(0)) == -1 {
+	if value.Cmp(zero) == -1 {
 		return nil, fmt.Errorf("cannot compute voting power on a negative value")
 	}
 
 	return big.NewInt(0).Sqrt(value), nil
 }
 
+// computeAccountLeveledPower takes a value and some voter data and returns the voting power of that value in
+//  the following way: the power of all votes combined has to be sqrt(sum(allAccountVotes)). So, the new
+//  vote will have a smaller power depending on how much existed previously
+func (g *governanceContract) computeAccountLeveledPower(value *big.Int, voteData *VoteSet) (*big.Int, error) {
+	totalBalance := big.NewInt(0)
+	for _, accountVote := range voteData.VoteItems {
+		if accountVote.Type != Account {
+			continue
+		}
+
+		totalBalance.Add(totalBalance, accountVote.Balance)
+	}
+
+	previousAccountPower, err := g.computeVotingPower(totalBalance)
+	if err != nil {
+		return nil, err
+	}
+
+	fullAccountBalance := big.NewInt(0).Add(totalBalance, value)
+	newAccountPower, err := g.computeVotingPower(fullAccountBalance)
+	if err != nil {
+		return nil, err
+	}
+
+	return big.NewInt(0).Sub(newAccountPower, previousAccountPower), nil
```

Comment:
This value greatly depend upon the last value. Suppose we have the value of 2500 as total balance and we add 2500 more, sqrt(2500 + 2500) - sqrt(2500) = 70 - 50 = 25. 
If we had the value of 1 as previous total balance, then the result would have been: sqrt(1 + 2500) - sqrt(1) = 50 - 1 = 49
Is this intended? If yes, why?

Reply:
Yes, this was intended.
First rule should be that an account's total power should always be sqrt(fullBalanceUsedToVote).
So now, in the case of multiple consecutive votes from the same address, any vote is very dependent and will have less effect depending on how much funds were previously used by that address.
This is not perfect, but mitigates something that would be even more strange (and undesirable I think), ex:
You make a vote with 25 so you know you have a power of 5 voted for `yes`. Then you make another vote (regardless of the value). If we would level all the votes proportionally, we would keep changing the past votes, so your vote you know you cast, would have a smaller value.

This way I find it more straight forward, if you want to make another vote, you know it will have a smaller power and you can determine what that power would be. You don't need to keep in mind how the new vote will affect your other votes.

Reply:
ok